bin/collator/src/parachain/shell_upgrade.rs

```diff
@@ -113,14 +113,25 @@ where
 {
     async fn verify(
         &mut self,
-        block_import: BlockImportParams<Block, ()>,
+        mut block_import: BlockImportParams<Block, ()>,
     ) -> Result<
         (
             BlockImportParams<Block, ()>,
             Option<Vec<(CacheKeyId, Vec<u8>)>>,
         ),
         String,
     > {
+        // Skip checks that include execution, if being told so or when importing only state.
+        //
+        // This is done for example when gap syncing and it is expected that the block after the gap
+        // was checked/chosen properly, e.g. by warp syncing to this block using a finality proof.
+        // Or when we are importing state only and can not verify the seal.
+        if block_import.with_state() || block_import.state_action.skip_execution_checks() {
```

Comment:
https://github.com/paritytech/substrate/issues/13921#issuecomment-1518812972
This is for shiden.